homeassistant/components/lovelace/__init__.py

```diff
@@ -120,21 +152,27 @@ def save_yaml(fname: str, data: JSON_TYPE):
                 _LOGGER.error("YAML replacement cleanup failed: %s", exc)
 
 
-def _yaml_unsupported(loader, node):
+def _yaml_unsupported(constructor, node):
     raise UnsupportedYamlError(
         'Unsupported YAML, you can not use {} in ui-lovelace.yaml'
         .format(node.tag))
 
 
-def load_yaml(fname: str) -> JSON_TYPE:
+def load_yaml(fname: str, rt: bool) -> JSON_TYPE:
     """Load a YAML file."""
     from ruamel.yaml import YAML
-    from ruamel.yaml.constructor import RoundTripConstructor
+    from ruamel.yaml.constructor import SafeConstructor
     from ruamel.yaml.error import YAMLError
 
-    RoundTripConstructor.add_constructor(None, _yaml_unsupported)
-
-    yaml = YAML(typ='rt')
+    if rt:
+        yaml = YAML(typ='rt')
+        yaml.preserve_quotes = True
+    else:
+        SafeConstructor.add_constructor(u'!secret', _secret_yaml)
```

Comment:
I am confused, how often will this be added to SafeConstructor, on every invocation of this function?

Reply:
I also think that this code belongs in its own PR